app/controllers/api/v1/upvote_controller.rb

```diff
@@ -0,0 +1,16 @@
+class Api::V1::UpvoteController < ApplicationController
+  skip_before_action :verify_autheticity_token
+
+  def create
+    upvote = JSON.parse(request.body.read)
```

Comment:
In an ideal world, you shouldn't need to `JSON.parse()` the request body this way. You should instead be able to let the Rails `params` magic take care of all of that for you, if everything is correctly-formulated.